src/pycountry/db.py

```diff
@@ -92,6 +99,45 @@ def _load(self):
 
     # Public API
 
```

Comment:
The following lines are obviously the core of this PR and both functions follow a similar 3-step approach:
1. Find/create the object in question
2. Modify self.objects
3. Update indices

The part about updating indices is a bit repetitive, but I wasn't entirely sure how to best make this generic, other than building the entire cache from scratch every time a change is made. Ideas welcome!